exchange/service.go

```diff
@@ -679,23 +685,38 @@ func ServiceDefResolver(wURL string, wOrg string, wVersion string, wArch string,
 				// Make sure the required service has the same arch as the service.
 				// Convert version to a version range expression (if it's not already an expression) so that the underlying GetService
 				// will return us something in the range required by the service.
+				var serviceDef *ServiceDefinition
 				if sDep.Arch != wArch {
-					return nil, nil, "", errors.New(fmt.Sprintf("service %v has a different architecture than the top level service.", sDep))
+					return nil, nil, nil, "", errors.New(fmt.Sprintf("service %v has a different architecture than the top level service.", sDep))
 				} else if vExp, err := semanticversion.Version_Expression_Factory(sDep.Version); err != nil {
-					return nil, nil, "", errors.New(fmt.Sprintf("unable to create version expression from %v, error %v", sDep.Version, err))
-				} else if s_map, s_def, s_id, err := ServiceDefResolver(sDep.URL, sDep.Org, vExp.Get_expression(), sDep.Arch, serviceHandler); err != nil {
-					return nil, nil, "", err
+					return nil, nil, nil, "", errors.New(fmt.Sprintf("unable to create version expression from %v, error %v", sDep.Version, err))
+				} else if apiSpecs, s_map, s_def, s_id, err := ServiceDefResolver(sDep.URL, sDep.Org, vExp.Get_expression(), sDep.Arch, serviceHandler); err != nil {
+					return nil, nil, nil, "", err
 				} else {
 					service_map[s_id] = *s_def
 					for id, s := range s_map {
 						service_map[id] = s
 					}
+
+					// Add all service dependencies to the running list of API specs.
+					serviceDef = s_def
+					for _, as := range *apiSpecs {
+						// If the apiSpec is already in the list, ignore it by ignoring the returned error.
+						res.Add_API_Spec(&as)
+					}
 				}
+
+				// Capture the current service dependency as an API Spec object and add it to the running list of API specs.
```

Comment:
Since this function recursively checks all dependencies, isnt this section extraneous/un-necessary? Or maybe the section above is extraneous.

Reply:
This section captures the current the dependents. and the line 700-706 captures the dependent's dependent. I copied the code from the ServiceResolver().